code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -211,28 +229,36 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
 	removed.temperature = temperature
 	for(var/id in cached_gases)
-		ADD_GAS(id, removed.gases)
+		ADD_GAS(id, removed_gases)
 		removed_gases[id][MOLES] = QUANTIZE(cached_gases[id][MOLES] * ratio)
 		cached_gases[id][MOLES] -= removed_gases[id][MOLES]
 
 	garbage_collect()
 
+	removed.heat_capacity = ratio * heat_capacity
+	heat_capacity *= (1 - ratio)
```

Comment:
Why do the gas moles get QUANTIZED() here? This could cause the tracked heat capacity to slightly drift from the actual heat capacity.

Reply:
I am just guessing here, but I bet this is due to the game-ruining NaN BS that several coders have worked extremely hard to remove.

Reply:
we quantize anytime we move gas. the quantized amount is the smallest amount of movement we treat as "real".
This might cause a problem with the idea of tracking heat capacity you're right. might need to re-examine either quantize or the heatcap system